spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkParquetWriter.java

```diff
@@ -151,4 +152,27 @@ public void testFpp() throws IOException, NoSuchFieldException, IllegalAccessExc
       assertThat(fpp).isEqualTo(0.05);
     }
   }
+
+  @Test
+  public void testColumnStatsEnabled()
+      throws IOException, NoSuchFieldException, IllegalAccessException {
+    File testFile = File.createTempFile("junit", null, temp.toFile());
+    try (FileAppender<InternalRow> writer =
+        Parquet.write(Files.localOutput(testFile))
+            .schema(SCHEMA)
+            .set(PARQUET_COLUMN_STATS_ENABLED_PREFIX + "id_long", "false")
+            .createWriterFunc(
+                msgType ->
+                    SparkParquetWriters.buildWriter(SparkSchemaUtil.convert(SCHEMA), msgType))
+            .build()) {
+      // Using reflection to access the private 'props' field in ParquetWriter
+      Field propsField = writer.getClass().getDeclaredField("props");
+      propsField.setAccessible(true);
+      ParquetProperties props = (ParquetProperties) propsField.get(writer);
+      MessageType parquetSchema = ParquetSchemaUtil.convert(SCHEMA, "test");
+      ColumnDescriptor idlDescriptor = parquetSchema.getColumnDescription(new String[] {"id_long"});
+      // Default statisticsEnabled should be true and for column id_long, it is disabled.
+      assertThat(props.getStatisticsEnabled(idlDescriptor)).isEqualTo(false);
+    }
+  }
```

Comment:
Why is this test in Spark?
The test is testing a Parquet writer feature. I think it should be in TestParquet, or somewhere near that. In this case we don't need reflection to test the setting. Package-private method, and VisibleForTesting annotation would be enough.

Also it would be good to test the actual effect on the Parquet files if that is possible.

I see that the previous test for bloom filter is done here, but that means that should have done there too. Maybe move it in a different PR?

Reply:
Also could we check that the default setting is working as well?

Reply:
Thanks @pvary for review. I will read more to see if the test can be moved to TestParquet and if test on the parquet files. 
As for default, there is a bug in Parquet-mr module with default, https://github.com/apache/parquet-java/pull/3189, right now, the default is always true which means it cannot be really configured.
My intention is that:
  1. User can configure default to true, and disable stats for specific columns. or
  2. User can configure default to false, enable stats for specific columns.

Reply:
We need to add this to the documentation then

Reply:
Ack

Reply:
I moved the unitest to TestParquet. The default test is not added as it depends on parquet-mr module.